ambry-store/src/test/java/com/github/ambry/store/BlobStoreCompactorTest.java

```diff
@@ -1617,9 +1621,9 @@ public void undeleteSameIndexSegmentTest_NoTtlUpdate() throws Exception {
       compactor.close(0);
     }
     // Before compaction, the records in the log are
-    // P11 U11 P P P| P12 D12 U12 U12 P
+    // P11 U11 P P P| P12 D12 U12 U12 P | P13(2) P P P P
```

Comment:
Could you explain what the `P P P` are? (other PUT records?). Also why last 4 "P"s get removed after compaction? (Are they duplicate PUT records ?)

Reply:
they are just random PUTs. The reason we have random puts here, is to make sure each index segment deal with one case of compaction. And the last 4 PUTs are not compacted. I just didn't bother to put it here since they are random PUTs.